2012-12-25
1 国际化分析,因为auth的form出错信息不知道为什么会显示中文,于是看了下源代码
	~/.virtualenvs/django_dev_01/lib/python2.7/site-packages/django/contrib/auth/forms.py


    error_messages = {
        'invalid_login': _("Please enter a correct username and password. "
                           "Note that both fields are case-sensitive."),
        'no_cookies': _("Your Web browser doesn't appear to have cookies "
                        "enabled. Cookies are required for logging in."),
        'inactive': _("This account is inactive."),
    }
定义了错误信息,前面用了 "_"符号,这是一个短别名,再前面引用了from django.utils.translation import ugettext, ugettext_lazy as _
使用函数 ugettext() 指定一个待翻译的字符串

        if username and password:
            self.user_cache = authenticate(username=username,
                                           password=password)
            if self.user_cache is None:
                raise forms.ValidationError(
                    self.error_messages['invalid_login'])
            elif not self.user_cache.is_active:
                raise forms.ValidationError(self.error_messages['inactive'])
        self.check_for_test_cookie()
        return self.cleaned_data

使用的时候调用error_message相应的信息,

建立文件
mkdir  locale
django-admin.py makemessages -l zh_CN


然后有下面一段
#: forms.py:143
msgid ""
"Please enter a correct username and password. Note that both fields are case-"
"sensitive."
msgstr "请输入正确的用户名和密码。请注意两者都是大小写敏感的。"

143行就是上面的'invalid_login': _("Please enter a correct username and password. "
                           "Note that both fields are case-sensitive."),


然后执行django-admin.py compilemessages


2 我的国际化
	国际化参考
	http://www.ibm.com/developerworks/cn/web/1101_jinjh_djangoi18n/
	https://docs.djangoproject.com/en/1.4/topics/i18n/translation/
	mkdir  locale  
	django-admin.py makemessages -l zh_CN  
	报错 xgettext: command not found
	安装gettext
	brew install gettext
	brew link gettext